python/ray/tests/test_client_proxy.py

```diff
@@ -0,0 +1,88 @@
+import os
+import pickle
+import pytest
+import time
+
+import grpc
+import ray
+import ray.core.generated.ray_client_pb2 as ray_client_pb2
+from ray.job_config import JobConfig
+import ray.util.client.server.proxier as proxier
+
+
+def test_proxy_manager(shutdown_only):
+    ray_instance = ray.init()
+    proxier.CHECK_PROCESS_INTERVAL_S = 1
+    os.environ["TIMEOUT_FOR_SPECIFIC_SERVER_S"] = "5"
+    pm = proxier.ProxyManager(ray_instance["redis_address"],
+                              ray_instance["session_dir"])
+    pm._free_ports = [45000, 45001]
+    client = "client1"
+
+    pm.start_specific_server(client, JobConfig())
+    # Channel should be ready and corresponding to an existing server
+    grpc.channel_ready_future(pm.get_channel(client)).result(timeout=5)
+
+    proc = pm._get_server_for_client(client)
+    assert proc.port == 45000
+
+    proc.process_handle().process.wait(5)
+    # Wait for reconcile loop
+    time.sleep(2)
+
+    assert len(pm._free_ports) == 2
+    assert pm._get_unused_port() == 45001
+
+
+@pytest.mark.parametrize(
+    "call_ray_start",
+    ["ray start --head --ray-client-server-port 25001 --port 0"],
+    indirect=True)
+def test_multi_clients(call_ray_start):
```

Comment:
what are we testing here? what the TODO means?

Reply:
Added, but ensuring that JobIDs are separate!